### PR TITLE
Implement CreateTodo and GetAllTodos on Frontend and Backend

### PR DESCRIPTION
**Pull request by maikmb**
- Implement CreateTodo and GetAllTodos methods on frontend using Apollo React Hooks
- Implement GetList usecase to execute without ids filter on backend

### CODE REVIEW

**Comment by jhomarolo** [comment]
Hello Maik, your PR is breaking our build and is also dramatically reducing test coverage.

Can you correct these points please?

**Comment by maikmb** [comment]
I increased the test coverage and now everything is fine!

image

File: backend/src/domain/usecases/getLists.js [hunk 1]
```diff
@@ -14,6 +14,14 @@ module.exports.getLists = injection =>
 
     'Get lists': step(async ctx => {
       const listRepo = new ctx.di.ListRepository(injection)
-      return (ctx.ret = await listRepo.getByIDs(ctx.req.ids))
+
+      if (ctx.req.ids.length > 0) {
```
Comment (jhomarolo) [inline backend/src/domain/usecases/getLists.js:18]: Here we should an ifelse use case

Something like this:

```
           "Get itens by Id or get all": ifElse({

            "If the id exists": step(async (ctx) => {
                if (ctx.req.ids.length > 0) {
                return Ok()
                }
            }),

            "Then: Get By Id": step(async (ctx) => {
                return await listRepo.getByIDs(ctx.req.ids)
            }),

            "Else: Get All": step(async (ctx) => {
                return await listRepo.getAll()
            })
        })
```

File: frontend/package.json [hunk 2]
```diff
@@ -21,7 +22,7 @@
   },
   "devDependencies": {
     "babel-eslint": "^10.1.0",
-    "eslint": "^7.4.0",
+    "eslint": "^6.6.0",
```
Comment (jhomarolo) [inline frontend/package.json:25]: Why did you decrease the version of eslint? There is a reason for this?

File: backend/src/domain/usecases/getLists.test.js [hunk 3]
```diff
@@ -24,6 +24,25 @@ describe('Get Todo Lists', () => {
...
+            const user = { canGetLists: true }
+            const req = { ids: [1, 2] }
+
+            // When
+            const uc = getLists(injection)
+            uc.authorize(user)
+            const ret = await uc.run({ ids: req.ids })
+
+            // Then
+            assert.ok(ret.isOk)
```
Comment (jhomarolo) [inline backend/src/domain/usecases/getLists.test.js:44]: Here, in addition to testing if "isOk", we could improve this test by checking the content to see if it's what we expect

File: backend/src/domain/usecases/getLists.js (hunk 1 above)
Comment (maikmb) [reply backend/src/domain/usecases/getLists.js]: Resolved!

File: frontend/package.json (hunk 2 above)
Comment (maikmb) [reply frontend/package.json:25]: Resolved!

File: backend/src/domain/usecases/getLists.test.js (hunk 3 above)
Comment (maikmb) [reply backend/src/domain/usecases/getLists.test.js:44]: Resolved!

**Comment by maikmb** [comment]
> Hello Maik, I suggested some small changes, thanks for your PR

Hi @jhomarolo! I fixed some files.
Can you approve, please?

**Review by jhomarolo-vortx — APPROVED** [review]

**Pull request merged**